src/utilities/control_panel.py

```diff
@@ -0,0 +1,105 @@
+from typing import Union
+
+from pyautogui import keyDown, keyUp
+
+import utilities.imagesearch as imsearch
+from model.osrs.osrs_bot import OSRSBot
+from utilities.random_util import random_chance
+
+
+class InvOpen(object):
+    def __init__(self, osrs_bot: OSRSBot):
+        self.osrs_bot = osrs_bot
+
+    def __enter__(self):
+        return _ensure_inventory_open(self.osrs_bot)
+
+    def __exit__(self, *args, **kwargs):
+        pass
+
+
+class MagicOpen(object):
+    def __init__(self, osrs_bot: OSRSBot):
+        self.osrs_bot = osrs_bot
+
+    def __enter__(self):
+        return _ensure_magic_tab_open(self.osrs_bot)
+
+    def __exit__(self, *args, **kwargs):
+        pass
+
+
+def get_active_tab(osrs_bot: OSRSBot) -> Union[int, None]:
```

Comment:
I made this function public since it seems like it would be useful to call directly from within a bot script. Now that I look at it, most of these functions could also be like that as well... Is it necessary to have a class for Inv and Magic when the developer could easily do something like:
```python
import utilities.control_panel as cp_util
if not cp_util.is_tab_active(self, 3):
    # Inv is not open, so open it
    self.mouse.move_to(self.win.cp_tabs[3].random_point())
    self.mouse.click()
```

Or even a function called `ensure_tab_open(bot: Bot, tab: int) -> bool:` that works for any tab?